rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1669,10 +1663,39 @@ public BooleanWrapper shutdown(boolean preempt) {
                 removeAllNodes(entry.getKey(), preempt, true);
                 entry.getValue().shutdown(caller);
             }
+            waitForAllNodeSourcesToBeShutdown();
+            finalizeShutdown();
         }
         return new BooleanWrapper(true);
     }
 
+    private void waitForAllNodeSourcesToBeShutdown() {
+        boolean atLeastOneAlive = false;
+        int millisBeforeHardShutdown = 0;
+        try {
+            do {
+                millisBeforeHardShutdown++;
+                Thread.sleep(100);
+                for (Entry<String, NodeSource> entry : this.deployedNodeSources.entrySet()) {
+                    atLeastOneAlive = atLeastOneAlive || isNodeSourceAlive(entry);
+                }
+            } while (atLeastOneAlive &&
+                     millisBeforeHardShutdown < PAResourceManagerProperties.RM_SHUTDOWN_TIMEOUT.getValueAsInt() * 10);
+        } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)